openstudiocore/src/model/ZoneVentilationDesignFlowRate.cpp

```diff
@@ -259,27 +261,84 @@ namespace detail {
   }
 
   bool ZoneVentilationDesignFlowRate_Impl::setDesignFlowRateCalculationMethod(std::string designFlowRateCalculationMethod) {
+    LOG(Warn, "ZoneVentilationDesignFlowRate::setDesignFlowRateCalculationMethod has been deprecated and will be removed in a future release, the design flow rate calculation method is set during the call to setDesignFlowRate, setFlowRateperZoneFloorArea, setAirChangesperHour, etc");
     bool result = setString(OS_ZoneVentilation_DesignFlowRateFields::DesignFlowRateCalculationMethod, designFlowRateCalculationMethod);
     return result;
   }
 
   bool ZoneVentilationDesignFlowRate_Impl::setDesignFlowRate(double designFlowRate) {
-    bool result = setDouble(OS_ZoneVentilation_DesignFlowRateFields::DesignFlowRate, designFlowRate);
+    bool result = true;
+    if (designFlowRate < 0){
+      result = false;
+    }else{
+      result = setString(OS_ZoneVentilation_DesignFlowRateFields::DesignFlowRateCalculationMethod, "Flow/Zone");
+      OS_ASSERT(result);
+      result = setDouble(OS_ZoneVentilation_DesignFlowRateFields::DesignFlowRate, designFlowRate);
+      OS_ASSERT(result);
+      result = setDouble(OS_ZoneVentilation_DesignFlowRateFields::FlowRateperZoneFloorArea, 0.0);
```

Comment:
@jmarrec technically you wouldn't want to set these fields if the previous 

`result = setDouble(OS_ZoneVentilation_DesignFlowRateFields::DesignFlowRate, designFlowRate);`

had failed (e.g. because the user passed in NaN, that is just fresh in my brain now).

Reply:
Otherwise looks good

Reply:
Good point. I did the setDouble stuff a month after this. I pushed another commit to address this.